### PR TITLE
Replaced ignoredGroups with ignoredPatterns

### PR DESCRIPTION
**Pull request by mudkiplex**
* Checks each full artifact specifier for the given patterns
via simple contains condition and ignores if a match is found.
* Adapted and added tests.
* Updated README.md.

### CODE REVIEW

File: gradle-license-plugin/src/test/groovy/com/jaredsburrows/license/LicensePluginAndroidSpec.groovy [hunk 1]
```diff
@@ -1,16 +1,13 @@
 package com.jaredsburrows.license
 
-import static org.gradle.testkit.runner.TaskOutcome.SUCCESS
-import static test.TestUtils.assertHtml
```
Comment (jaredsburrows) [inline gradle-license-plugin/src/test/groovy/com/jaredsburrows/license/LicensePluginAndroidSpec.groovy:4]: Undo this?

File: gradle-license-plugin/src/test/groovy/com/jaredsburrows/license/LicensePluginAndroidSpec.groovy [hunk 2]
```diff
@@ -1,16 +1,13 @@
...
-import static test.TestUtils.gradleWithCommand
-import static test.TestUtils.getLicenseText
-
 import org.junit.Rule
 import org.junit.rules.TemporaryFolder
 import spock.lang.Specification
 import spock.lang.Unroll
 
+import static org.gradle.testkit.runner.TaskOutcome.SUCCESS
+import static test.TestUtils.*
```
Comment (jaredsburrows) [inline gradle-license-plugin/src/test/groovy/com/jaredsburrows/license/LicensePluginAndroidSpec.groovy:9]: Please do not import * imports.

**Comment by jaredsburrows** [comment]
Thanks!

File: gradle-license-plugin/src/test/groovy/com/jaredsburrows/license/LicensePluginAndroidSpec.groovy (hunk 1 above)
Comment (mudkiplex) [reply gradle-license-plugin/src/test/groovy/com/jaredsburrows/license/LicensePluginAndroidSpec.groovy:4]: Android studio did this for me via optimise imports. Fixed!

**Comment by mudkiplex** [comment]
@jaredsburrows Could you publish these changes?

**Pull request merged**